File: services/python/fastapi/src/application.py
import asyncio
from contextlib import asynccontextmanager
from fastapi import FastAPI
from fastapi.exceptions import RequestValidationError

from app.state import AppState
from routes.middleware import log_middleware, chaos_middleware, body_size_middleware
from routes.router import register_routes
from routes.handlers import (
    internal_handler,
    invalid_method_handler,
    invalid_path_handler,
    validation_handler,
)


def create_app() -> FastAPI:
    app = FastAPI(lifespan=lifespan)

    app.middleware("http")(body_size_middleware)
    app.middleware("http")(log_middleware)
    app.middleware("http")(chaos_middleware)

    app.add_exception_handler(RequestValidationError, validation_handler)
    app.add_exception_handler(404, invalid_path_handler)
    app.add_exception_handler(405, invalid_method_handler)
    app.add_exception_handler(Exception, internal_handler)

    register_routes(app)

    return app


@asynccontextmanager
async def lifespan(app: FastAPI):
    app.state.app_state = AppState()
    app.state.tasks_lock = asyncio.Lock()
    yield

File: services/go/gin/application.go
package main

import (
	"app"
	"gin/routes/handlers"
	"gin/routes/middleware"

	"github.com/gin-gonic/gin"
)

type Application struct {
	Engine *gin.Engine
	state  *app.AppState
}

func NewApplication(appConfig app.AppConfig) *Application {
	state := app.NewState(appConfig)

	gin.SetMode(gin.ReleaseMode)

	engine := gin.New()
	engine.HandleMethodNotAllowed = true
	engine.SetTrustedProxies(nil)
	engine.Use(gin.CustomRecovery(handlers.InternalHandler))
	engine.Use(middleware.BodySizeMiddleware())
	engine.Use(middleware.LogMiddleware())
	engine.Use(middleware.ChaosMiddleware(state))

	engine.POST("/", handlers.PostHandler(state))
	engine.GET("/:id", handlers.GetHandler(state))
	engine.DELETE("/:id", handlers.DeleteHandler(state))
	engine.PATCH("/:id", handlers.PatchHandler(state))
	engine.PUT("/:id", handlers.PutHandler(state))

	engine.NoRoute(handlers.InvalidPathHandler)
	engine.NoMethod(handlers.InvalidMethodHandler)

	return &Application{Engine: engine, state: state}
}
